addons/website_sale_slides/models/product_product.py

```diff
@@ -9,6 +9,12 @@ class Product(models.Model):
 
     channel_ids = fields.One2many('slide.channel', 'product_id', string='Courses')
 
+    def _is_add_to_cart_allowed(self):
```

Comment:
Hello @Feyensv 

Did anything change in this regard recently? I'm not sure why this pops up in 17.4.

I see that for events, we added published events to the list of allowed products.
I'm guessing we could apply the same idea here and filter on published courses?
Or is there another trick to this?

Thanks & cheers!

Reply:
Yes there was an feature to restrict shop visibility depending on setting and because of that only publish product (related to courses) will not allowed so we need this override to bypass that check for product related to courses same as event and others

[See condition has_ecommerce_access](https://github.com/odoo/odoo/pull/152088/files#diff-bcce8086c8655fe6671ccbb5b4739a891993e92380f67930ea24423e80941607R138) which is preventing adding course related product into cart even though they are published

Have a nice day!